harmonica/forward/point_mass.py

```diff
@@ -303,8 +287,8 @@ def kernel_g_r(
     """
     Kernel function for radial component of gravity gradient in spherical coordinates
     """
-    coslambda = np.cos(longitude_p - longitude)
-    cospsi = sinphi_p * sinphi + cosphi_p * cosphi * coslambda
-    distance_sq = (radius - radius_p) ** 2 + 2 * radius * radius_p * (1 - cospsi)
+    distance_sq, cospsi, _ = _distance_sq_spherical_core(
+        longitude, cosphi, sinphi, radius, longitude_p, cosphi_p, sinphi_p, radius_p
+    )
     delta_z = radius_p * cospsi - radius
     return delta_z / distance_sq ** (3 / 2)
```

Comment:
I learned a while ago that sometimes this is actually a lot slower than `np.sqrt(distance)**3`. So we don't really need the distance to be squared.

Reply:
:+1: Didn't know that. I thought that applying a single power to a value would be faster than applying two different ones in series. Although I have no references to hold this statement, just intuition.